Maple2.Model/Game/Market/PremiumMarketItem.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections.Generic;
+using Maple2.Model.Enum;
+using Maple2.Model.Metadata;
+using Maple2.PacketLib.Tools;
+
+namespace Maple2.Model.Game;
+
+public class PremiumMarketItem : MarketItem {
+    public int Id { get; init; }
```

Comment:
everywhere you're using `init`, you can also use `required` to ensure values are set when constructing the object.
```cs
public required int Id { get; init; }
```